Public/Src/Cache/ContentStore/Distributed/Sessions/DistributedContentSession.cs

```diff
@@ -165,5 +180,51 @@ private async Task<PutResult> RegisterPutAsync(Context context, CancellationToke
 
             return putResult;
         }
+
+        private async Task<BoolResult> RequestProactiveCopyIfNeededAsync(OperationContext context, ContentHash hash)
+        {
+            var hashArray = new[] { hash };
+
+            // First check in local location store, then global if failed.
+            var getLocationsResult = await ContentLocationStore.GetBulkAsync(context, hashArray, context.Token, UrgencyHint.Nominal, GetBulkOrigin.Local);
+            if (!getLocationsResult.Succeeded || getLocationsResult.ContentHashesInfo[0].Locations.Count == 0)
+            {
+                getLocationsResult = await ContentLocationStore.GetBulkAsync(context, hashArray, context.Token, UrgencyHint.Nominal, GetBulkOrigin.Global);
+
+                if (getLocationsResult.Succeeded)
+                {
+                    _counters[Counters.GetLocationsSatisfiedFromRemote].Increment();
+                }
+            }
+            else
+            {
+                _counters[Counters.GetLocationsSatisfiedFromLocal].Increment();

```

Comment:
nit, but I would suggest to reverse the if and do:

if (getLocationsResult && getLocationsResult.LocationsCount != 0) {
  counters[Local].Increment();
}